aptos-move/framework/supra-stdlib/sources/enumerable_map.move

```diff
@@ -122,57 +123,93 @@ module supra_std::enumerable_map {
     }
 
     /// Will clear the entire data from the Enumerable Map
-    public fun clear<K : copy+drop, V : store+drop+copy>(map: &mut EnumerableMap<K, V>) {
+    public fun clear<K: copy+drop, V: store+drop+copy>(map: &mut EnumerableMap<K, V>) {
         let list = get_map_list(map);
         remove_value_bulk(map, list);
     }
 
     /// Returns the value of a key that is present in Enumerable Map
-    public fun get_value<K : copy+drop, V : store+drop+copy>(map: & EnumerableMap<K, V>, key: K): V {
+    public fun get_value<K: copy+drop, V: store+drop+copy>(map: & EnumerableMap<K, V>, key: K): V {
         table::borrow(&map.map, key).value
     }
 
     /// Returns reference to the value of a key that is present in Enumerable Map
-    public fun get_value_ref<K : copy+drop, V : store+drop+copy>(map: & EnumerableMap<K, V>, key: K): &V {
+    public fun get_value_ref<K: copy+drop, V: store+drop+copy>(map: & EnumerableMap<K, V>, key: K): &V {
         &table::borrow(&map.map, key).value
     }
 
 
     /// Returns the value of a key that is present in Enumerable Map
-    public fun get_value_mut<K : copy+drop, V : store+drop+copy>(map: &mut EnumerableMap<K, V>, key: K): &mut V {
+    public fun get_value_mut<K: copy+drop, V: store+drop+copy>(map: &mut EnumerableMap<K, V>, key: K): &mut V {
         &mut table::borrow_mut(&mut map.map, key).value
     }
 
     /// Returns the list of keys that the Enumerable Map contains
-    public fun get_map_list<K : copy+drop, V : store+drop+copy>(map: &EnumerableMap<K, V>): vector<K> {
+    public fun get_map_list<K: copy+drop, V: store+drop+copy>(map: &EnumerableMap<K, V>): vector<K> {
         return map.list
     }
 
     /// Check whether Key is present into the Enumerable map or not
-    public fun contains<K: copy+drop, V : store+drop+copy>(map: &EnumerableMap<K, V>, key: K): bool {
+    public fun contains<K: copy+drop, V: store+drop+copy>(map: &EnumerableMap<K, V>, key: K): bool {
         table::contains(&map.map, key)
     }
 
     /// Return current length of the EnumerableSetRing
-    public fun length<K : copy+drop, V : store+drop+copy>(set: &EnumerableMap<K, V>): u64 {
+    public fun length<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>): u64 {
         return vector::length(&set.list)
     }
 
+    /// Apply the function to each element in the EnumerableMap.
+    public inline fun for_each_value<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>, f: |V|) {
+        let i = 0;
+        let len = length(set);
+        while (i < len) {
+            let key = *vector::borrow(&set.list, i);
+            f(table::borrow(&set.map, key).value);
+            i = i + 1
+        }
+    }
```

Comment:
I think we also need `for_each_value_mut` also where a mutable reference of the `value` is passed to the function.

Reply:
We should also add `for_each_keyval`, where copies of `key` and `value` is provided to function (original data structure must not change).